src/ui/components/utils/notebook-snapshot.js

```diff
@@ -0,0 +1,135 @@
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+function generateTaskForm(url) {
+    var taskForm = {
+        name: "Create a Notebook Entry",
+        hint: "Please select a Notebook",
+        sections: [{
+            rows: [
+                {
+                    name: 'Entry',
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
+            size: imageSize
+        };
+
+        options.notebook.useCapability('mutation', function (model) {
+            var date = Date.now();
+
+            model.entries.push({
+                id: 'entry-' + date,
+                createdOn: date,
+                text: options.entry,
+                embeds: [{
+                    name: embedObject.name,
+                    cssClass: embedObject.cssClass,
+                    type: embedObject.id,
+                    id: 'embed-' + date,
+                    createdOn: date,
+                    snapshot: snapshotObject
+                }]
+            });
+        });
+    });
+}
+
+class NotebookSnapshot {
+    constructor(openmct) {
+        this.openmct = openmct;
+
+        this.capture = this.capture.bind(this);
+    }
+
+    capture(domainObject, DomElement) {
+        let exportImageService = this.openmct.$injector.get('exportImageService'),
```

Comment:
these should be in the constructor and exposed as member variables. The private functions included above should become member functions (annotated with @private jsdoc). We are standardizing on naming private functions with an underscore, eg. `_saveSnapShot`. JS will eventually support private members in classes properly via the [ghastly `#` syntax](https://github.com/tc39/proposal-private-methods#private-methods-and-fields).

With dialogService as a member variable, you won't need to pass it to `saveSnapshot`

Reply:
Done, makes sense. I always liked the idea of private functions.